cli/Sources/TuistServer/Utilities/CachedValueStore.swift

```diff
@@ -25,13 +31,32 @@ actor CachedValueStore: CachedValueStoring {
         }
     }
 
-    private var tasks: [String: Task<Any, any Error>] = [:]
+    private var tasks: [String: Task<Any?, any Error>] = [:]
     private var cache: [String: Any] = [:]
+    private let fileSystem = FileSystem()
+
+    /// Returns the path to the lock file for a given key
+    private func lockFilePath(for key: String) -> Path.AbsolutePath {
+        // Use a sanitized version of the key for the filename
+        let sanitizedKey = key.replacingOccurrences(of: "/", with: "_")
+            .replacingOccurrences(of: ":", with: "_")
+            .replacingOccurrences(of: " ", with: "_")
+
+        #if canImport(TuistSupport)
+            return Environment.current.stateDirectory
+                .appending(component: "cached_value_store")
+                .appending(component: "\(sanitizedKey).lock")
+        #else
+            fatalError(
+                "We need to figure out a file-locking solution that's not tied to Environment, or extract Environment from TuistSupport and make it work across platforms."
+            )
```

Comment:
wait – won't this crash in the macOS app? Have you tried to run it?

Reply:
Agree that's it out-of-scope. As long as this doesn't cause issues in the macOS app, we should be good.